mailpoet/assets/js/src/newsletter_editor/behaviors/TextEditorBehavior.ts

```diff
@@ -82,14 +80,14 @@ BL.TextEditorBehavior = Marionette.Behavior.extend({
 
         plugins: this.options.plugins,
 
-        setup: function setup(editor) {
+        setup: (editor) => {
           // Store the editor instance
-          that.tinymceEditor = editor;
-          editor.on('change', function onChange() {
-            that.view.triggerMethod('text:editor:change', editor.getContent());
+          this.tinymceEditor = editor;
+          editor.on('change', () => {
```

Comment:
Does converting the file to TS require replacing `function foo() {}` with `() => {}` or do we follow some standard to use the latter instead of the former?

I don't think you need to change this, but I'm asking mostly because I'm not a fan of the arrow function syntax in most cases as it makes it a bit harder to read the code in my opinion. My impression is that any developer, regardless of their programming language background, can easily understand `function foo() {}` or even `function() {}`, but the same is not true for arrow functions.

Reply:
converting to ts doesn't directly require converting these functions to arrow function, but in the former code were defining [that](https://github.com/mailpoet/mailpoet/pull/4610/files#diff-ef944cf4cc1566facf2c6d659dacb51dd061c3630a8c9ca7c9c40ccbabd590a5L46)  to be able to use the `this` from the upper context, but this kind of context binding is not allowed in eslint rules defined for our ts files, so I had two options:
1) Use arrow function, since arrow functions automatically are bound to their parent
2) Keep these functions and bind context using `bind(this)`
I chose the first option since it's more concise and ES6+ way of binding to the parent context.
Arrow functions are now more or less the new norm, at least in ES6+ JS world, the only justification is not using them can be that we can name the older function syntax, but we miss this possibility when using arrow functions.

Reply:
Thanks for taking the time to provide the additional context and for explaining why you opted to use arrow functions.